From f7e22577c5679970c28cc9fa6a2224f506a0a58c Mon Sep 17 00:00:00 2001
From: Xiang Xu <xuxiang@eswincomputing.com>
Date: Tue, 20 May 2025 17:14:45 +0800
Subject: [PATCH 512/652] feat:sbc support i2c, spi

Changelogs:
1. sbc support spi flash winbond,w25q128
2. sbc support ti,ina226,ti,tmp102
3. sbc support uart3

Signed-off-by: Xiang Xu <xuxiang@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7700-sbc-a1.dts | 42 +++++++++++++++-----
 1 file changed, 32 insertions(+), 10 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-sbc-a1.dts b/arch/riscv/boot/dts/eswin/eic7700-sbc-a1.dts
index 8dc5eb4c4607..e019b70ecda0 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-sbc-a1.dts
+++ b/arch/riscv/boot/dts/eswin/eic7700-sbc-a1.dts
@@ -561,7 +561,9 @@ &d0_uart2 {
 
 &d0_uart3 {
 	/* pin header */
-	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3_default>;
+	status = "okay";
 };
 
 &d0_uart4 {
@@ -571,12 +573,23 @@ &d0_uart4 {
 
 &ssi0 {
 	/* pin header */
-	status = "disabled";
+	status = "okay";
+	num-cs = <1>;
+	spi-flash@0 {
+		compatible = "winbond,w25q128",
+					"jedec,spi-nor";
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
 };
 
 &ssi1 {
 	/* pin header */
-	status = "disabled";
+	status = "okay";
+
 };
 
 &bootspi {
@@ -674,12 +687,25 @@ &d0_i2c3 {
 
 &d0_i2c4 {
 	/* pin header */
-	status = "disabled";
+	status = "okay";
+	sys_power:ina226@44 {
+		compatible = "ti,ina226";
+		#io-channel-cells = <1>;
+		label = "sys_power";
+		reg = <0x40>;
+		shunt-resistor = <1000>;
+	};
 };
 
 &d0_i2c5 {
 	/* pin header */
-	status = "disabled";
+	status = "okay";
+	tmp102@48 {
+		compatible = "ti,tmp102";
+		reg = <0x48>;
+		label = "board_tmp";
+		#thermal-sensor-cells = <1>;
+	};
 };
 
 &d0_i2c6 {
@@ -772,11 +798,7 @@ &pinctrl {
 	pinctrl-0 = <&pinctrl_gpio1_default &pinctrl_gpio2_default &pinctrl_gpio3_default &pinctrl_gpio4_default
 			&pinctrl_gpio5_default &pinctrl_gpio6_default &pinctrl_gpio7_default &pinctrl_gpio8_default
 			&pinctrl_gpio9_default &pinctrl_gpio10_default &pinctrl_gpio17_default &pinctrl_gpio27_default
-			&pinctrl_gpio35_default &pinctrl_gpio36_default &pinctrl_gpio37_default &pinctrl_gpio38_default
-			&pinctrl_gpio39_default &pinctrl_gpio40_default &pinctrl_gpio41_default &pinctrl_gpio52_default
-			&pinctrl_gpio53_default &pinctrl_gpio54_default &pinctrl_gpio55_default &pinctrl_gpio64_default
-			&pinctrl_gpio65_default &pinctrl_gpio66_default &pinctrl_gpio67_default &pinctrl_gpio92_default
-			&pinctrl_gpio93_default>;
+			&pinctrl_gpio64_default	&pinctrl_gpio65_default &pinctrl_gpio66_default &pinctrl_gpio67_default >;
 
     /* pin header default function for GPIO
        SPI1 (CS0,SCLK,MOSI,MISO,D2,D3,CS1): GPIO 35,36,37,38,39,40,41
-- 
2.52.0

